### DISCUSSION TITLE
SAE erhält JSON Werte, Hinzufügen in SHM Fehler

### DISCUSSION THREAD

**Discussion by runn8** [camueller/SmartApplianceEnabler#373] [Installation und Betrieb]
Hallo,

ich habe mich bemüht, die Dokumentation soweit möglich zu lesen. Trotzdem komme ich leider nicht weiter.
Im SAE waren bereits zwei Modbus Stromzähler integriert, die auch im SHM funktionieren.

Nun habe ich im SAE drei neue Shelly 1s Zwischenstecker ergänzt. Diese sind auch schön über http://<IP>/meter/0 aufrufbar und liefern Werte. Auch im SEMP XML-Payload des SAE sind die Werte enthalten. Im SHM werden die neuen (und ein bereits bestehendes?) Geräte zwar gefunden, ich kann sie jedoch nicht hinzufügen, s. Screenshot:
![grafik](https://user-images.githubusercontent.com/103494067/199756734-4c9f4868-7086-479f-a936-32801765f5e3.png)

In der [Q&A SEMP2](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md#semp2) habe ich gelesen, dass in der SEMP XML-Payload der <Status> in <DeviceStatus> nicht "Off" sein darf. Dieser ist jedoch bei den neuen Geräten (und sogar einem bisherigen Gerät) auf "off". Wie bringe ich diesen auf "On"? Interessanterweise findet der SHM als Neugerät auch das "Kochfeld", das aber bereits seit vielen Wochen erfolgreich eingebunden ist und auch funktioniert. Das Kochfeld hat übrigens auch den Status "Off" und funktioniert. Die ID ist selbstverständlich bei allen Geräten unterschiedlich.

Gerne würde ich auch die alten gefundenen Geräte des SAE aus dem SHM dauerhaft entfernen, aber das ist mir bislang nicht gelungen. Auch ein Neustart des SHM hat die verwaisten Einträge nicht gelöscht... Im SEMP-XML Payload sind die alten Einträge jedenfalls nicht mehr drin.

Danke für jeden Tipp!

**Comment by camueller** [comment]
> Nun habe ich im SAE drei neue Shelly 1s Zwischenstecker ergänzt. Diese sind auch schön über http:///meter/0 aufrufbar und liefern Werte. Auch im SEMP XML-Payload des SAE sind die Werte enthalten. Im SHM werden die neuen (und ein bereits bestehendes?) Geräte zwar gefunden, ich kann sie jedoch nicht hinzufügen, s. Screenshot: 

Der SHM redet ja nie direkt mit den Geräten (die vom SAE verwaltet werden), sondern nur mit dem SAE. Die Tatsache, dass er Dir die Geräte anzeigt deutet darauf hin, dass der SHM mit dem SAE kommunizieren kann bzw. konnte. Sind die angezigten Geräte tatsächlich diejenigen, welche der SAE jetzt gerade verwaltet oder sind das nur Überbleibsel aus der Vergangenheit? Um das herauszufinden kannst Du einfach einen Namen oder eine Seriennummer im SAE ändern und schauen, ob der SHM diese geänderten Werte bei den Geräten zum Hinzufügen anzeigt.

Ausser SHM reset habe ich keine Idee, wo die gezeigte Fehlermeldung ihre Ursache hat.

> In der [Q&A SEMP2](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/QA_DE.md#semp2) habe ich gelesen, dass in der SEMP XML-Payload der in nicht "Off" sein darf. Dieser ist jedoch bei den neuen Geräten (und sogar einem bisherigen Gerät) auf "off". Wie bringe ich diesen auf "On"? Interessanterweise findet der SHM als Neugerät auch das "Kochfeld", das aber bereits seit vielen Wochen erfolgreich eingebunden ist und auch funktioniert. Das Kochfeld hat übrigens auch den Status "Off" und funktioniert. 

Der Hinweis bezieht sich darauf, dass die Verbauchswerte nicht im Sunny Portal angezeigt werden, ist also in Deiner jetzigen Situation nicht relvant. Wenn Du einen Zähler konfiguriert hast, wird der Status automatisch richtig gesetzt, solange Du keinen Schaltet konfiguriert hast, der einen falschen Zustand meldet.

**Reply by runn8** [reply]
Vielen Dank für deine flotte Antwort, ich fühle mich geehrt! :-)

> Der SHM redet ja nie direkt mit den Geräten (die vom SAE verwaltet werden), sondern nur mit dem SAE. Die Tatsache, dass er Dir die Geräte anzeigt deutet darauf hin, dass der SHM mit dem SAE kommunizieren kann bzw. konnte. Sind die angezigten Geräte tatsächlich diejenigen, welche der SAE jetzt gerade verwaltet oder sind das nur Überbleibsel aus der Vergangenheit? Um das herauszufinden kannst Du einfach einen Namen oder eine Seriennummer im SAE ändern und schauen, ob der SHM diese geänderten Werte bei den Geräten zum Hinzufügen anzeigt.

Ich habe testweise verändert
Gerät1 habe ich belassen, funktioniert
Gerät2 Seriennummer+9 -> funktioniert im SHM, allerdings wird die alte Seriennummer angezeigt, obwohl die neue übergeben wurde. Das finde ich faszinierend. Oder nimmt der SHM die Device-ID als Referenzwert her und übermittelt Seriennummer u. Bezeichnung nur zur Information?

Gerät3 Seriennummer+9 -> wird im SHM als neues Gerät erkannt, allerdings nach wie vor mit alter Seriennummer
Gerät4 Seriennummer+9 -> wird im SHM als neues Gerät erkannt, allerdings nach wie vor mit alter Seriennummer
Gerät5 Seriennummer+9 -> wird im SHM als neues Gerät erkannt, allerdings nach wie vor mit alter Seriennummer

In der Semp-Ausgabe vom SAE sind alle neuen Seriennummern (mit Postfix "9") sauber enthalten.
Der SHM zeigt als gefundene Neugeräte mittlerweile 12 Geräte an, alle mit alten Seriennummern, aber nicht die neuen. Ich kann die obsoleten Geräte löschen, aber man kann im SHM nach wie vor "Gelöschte Neugeräte wiederherstellen". 

Ich vermute, der SHM hat hier irgendeinen Patschen. Wenn ich denselbe resette und nicht nur neu starte, sind dann sämtliche Geräteaufzeichnungen weg? Ich weiß nicht mehr weiter.

Noch etwas: Lt. SAE manual können nur 12 Geräte hinzugefügt werden inkl. Trick auf 24 Geräte. Scheinbar hat sich das geändert?
<img width="371" alt="grafik" src="https://user-images.githubusercontent.com/103494067/199825288-cc35bd92-1939-4187-8ed8-9ffe3073ff3e.png">


Freue mich über jeden Tipp! :-)

**Comment by camueller** [comment]
> Oder nimmt der SHM die Device-ID als Referenzwert her und übermittelt Seriennummer u. Bezeichnung nur zur Information?

Ja, so ist es.

> Ich vermute, der SHM hat hier irgendeinen Patschen. Wenn ich denselbe resette und nicht nur neu starte, sind dann sämtliche Geräteaufzeichnungen weg? Ich weiß nicht mehr weiter.

Solange Du ihn nicht auf Werkseinstellung zurücksetzt, sollte nichts passieren (da hab ich schoon öfters gemacht). Beim SHM 2.0 solltest Du den Reset-Knopf länger als 6 Sekunden gedrückt halten.

**Comment by runn8** [comment]
Hallo,

vielen Dank für deine Nachricht!
So, ich habe den SHM 2.0 sowohl neu gestartet (länger als 6s Reset drücken), sowie auch einen Reset durchgeführt (3-6s. Reset drücken). Die Einstellungen blieben bestehen. Ich habe sogar den SAE abgeschaltet, um zu sehen, welche Werte übermittelt werden. Nach wie vor findet der SHM eine Reihe von alten und zuletzt im SAE hinzugefügten Geräte:
<img width="749" alt="grafik" src="https://user-images.githubusercontent.com/103494067/200047953-5ef4c26b-f3e7-4b23-947d-7d7699ab98d2.png">

Woher hat der SHM diese Infos und warum vergisst er sie nicht? Dass diese selbst nach Neustart und Reset des SHM noch da sind, ist sehr ärgerlich.


Habe nun alle Geräte "gelöscht", aber man kann sie ja noch wiederherstellen:
<img width="802" alt="grafik" src="https://user-images.githubusercontent.com/103494067/200048386-5f8487b7-5921-4dbe-94f1-c880b3a66a90.png">

Was bleibt hier noch zu tun?

**Comment by camueller** [comment]
Der SHM hat diesbezüglich aktuell Probleme. Aber solange Du nur die Geräte hinzufügst, die Du auch aktuell im SAE konfiguriert hast, solltest ja funktionieren.

Ansonsten kann ich Dir nur empfehlen, verschiedene sinnvolle Dinge auszuprobieren, bis Du ein Setup hast, das für Dich funktioniert. Du kannst zwar versuchen, den SMA-Support deswegen zu kontaktieren, aber ich denke nicht, dass sie Dir weiterhelfen können.

**Comment by runn8** [comment]
Nach mehreren Versuchen hängte sich der SHM offenbar auf (rote LED blinkte). Ich habe diesen dann neu gestartet (Reset-Knopf für 7+ Sekunden gedrückt u. losgelasssen). Danach funktionierte das Portal wieder (aktuelle Werte) und auch der SAE wurde mit seinen Geräten gefunden. Zwar sind auch noch alle alten SAE-Einträge da, obwohl diese nicht über SEMP übertragen werden, aber zumindest funktionieren nun die neuen Geräte. Bei den SHM Parametern ist mir aufgefallen, dass der Wert "EEBUS SKI Gerätekennung" heute aktualisiert wurde. Möglicherweise hat das einen Einfluss auf die Kommunikation mit dem SAE...

Schönen Abend,
Mario
